dm/docs/RFCS/20211130_enhanced_pre_checker.md

```diff
@@ -0,0 +1,86 @@
+
+### Too much time overhead
+
+If we have a large number of tables in source, we will take too much time in checking table schema, sharding table consistency and sharding table auto increment ID.
+
+### Inadequate check
+
+* Now we check auto increment ID by `column-mapping` which is deprecated. If we don't set the `column-mapping` and don't ignore the **auto_increment_ID**, the pre-check will report an error. PS: **auto_increment_ID** is only checked when **schema_of_shard_tables** is true.
+* Dump privilege only checks RELOAD and SELECT. However, Dumpling supports different [consistency configurations](https://docs.pingcap.com/tidb/stable/dumpling-overview#adjust-dumplings-data-consistency-options), which need more privileges.
+* If online-ddl is set by true and a DDL is in online-ddl stage, DM will have a problem in all mode. Specifically, when ghost table has been created but not been renamed, DM will report an error when the ghost table is renamed during the incremental phase. You can learn more about online-ddl [here](https://docs.pingcap.com/tidb-data-migration/stable/feature-online-ddl).
+* For schema_of_shard_tables, whether it's a pessimistic task or an optimistic task, we all simply check consistency of schema by comparing all sharding tables’ structures. For optimistic mode, we can do better.
+* Version checker checks that MySQLVersion >= 5.6.0 and MariadbVersion >= 10.1.2 before. However, we find more and more incompatibility problems in MySQLVersion >= 8.0.0 and Mariadb. In view of supporting MySQL 8.0 and Mariadb is experimental yet, checker will report a warning for them.
+
+## Proposal
+
+### Speed ​​up check
+
+1. Support concurrent check
+    - table_schema
+    - schema_of_shard_tables
+    - auto_increment_ID
+2. We can adjust the concurrency by table numbers.
+
+### Optimize some check
+
+1. Auto_increment_ID only be checked in sharding mode.
+    - If table exists auto increment ID, report a warning to user and tell them the method that can resolve the PK/UK conflict;
+        1. If you set PK to AUTO INCREMENT, you must make sure that the primary key in sharding tables is not duplicated;
+        2. If sharding tables have duplicated PK, you can refer to [document](https://docs.pingcap.com/tidb-data-migration/stable/shard-merge-best-practices#handle-conflicts-of-auto-increment-primary-key).
+    - And if they have finished the resolving, such as manually created the table and removed AUTO INCREMENT, we should not report the warning.
+2. Dump_privilege will check different privileges according to different [consistency](https://docs.pingcap.com/tidb/stable/dumpling-overview#adjust-dumplings-data-consistency-options) on source.
+    - For all consistency, we will check
+        - SELECT (only INFORMATION_SCHEMA's tables and dump tables)
+    - For flush consistency:
+        - RELOAD (global)
+    - For flush/lock consistency:
+        - LOCK TABLES (only dump tables)
```

Comment:
After test, found that `flush tables with read lock` does not need this privilege. We juse used `flush tables with read lock`.😭

remove it.